api/services/auth.go

```diff
@@ -267,6 +268,56 @@ func (s *service) AuthGetToken(ctx context.Context, id string) (*models.UserAuth
 	}, nil
 }
 
+func (s *service) AuthCustomToken(ctx context.Context, id string, mfa bool, expiredAt time.Time) (map[string]interface{}, error) {
```

Comment:
To guarantee it works as expected, It's worth adding a test suit to this function cases.